geps/gep-1856.md

```diff
@@ -0,0 +1,48 @@
+# GEP 1856: Egress Route
+Issue: [#1856](https://github.com/kubernetes-sigs/gateway-api/issues/1856)
+
+Status: Provisional|Implementable|Experimental|Standard|Deferred|Rejected|Withdrawn|Replaced
+
+## TLDR
+Egress traffic is any traffic that originates inside a Kubernetes cluster that is bound for a resource outside of the cluster. Egress Traffic in Kubernetes is a common use case for Service Providers to provision outbound traffic that requires a Gateway resource to provide Source Network Address Translation (SNAT) capabilities.
+
+We propose a Gateway API Route allowing us to describe the path for an application to reach an endpoint outside of the cluster. The egress path is protocol agnostic and the Gateway can be connected to multiple external networks.
+
+## Goals
+- Provide a standardize configuration for provisioning egress for any protocol through a single Gateway.
+- Egress Gateways can be connected to different external networks.
+- Provision for using SNAT to allow for identification of cloud-native network functions (CNFs) via engineered IP assignments for security purposes.
+- Allow CNFs to be addressed by their external IPs regardless of whether they are being access from within the cluster or from outside the cluster.
+
+## Non-Goals
+Security functionality should be kept outside this "route". We can have these be referenced as appropriate.
+
+IP address management for SNATing will not duplicate non-trivial IPAM (IP address management) capabilities, only simple capability will be provided here.
+
+Potential future use cases such as:
+
+- Provision for using SNAT to allow for IPv4 to IPv6 translation on the way out of the cluster.
+- Allow CNFs using peering protocols like Diameter to function by allowing connections to originate in either direction.
+- Allow attaching policies (specifics to come later).
```

Comment:
Added more examples for Non-goals.

- Provision to allow for IPv4 to IPv6 translation on the way out of the cluster. _For example, maintaining a map of IPv4/IPv6 addresses to allow IPv4 internal resources to transparently access IPv6 external resources. In order to not overcomplicate the initial GEP, this can be postponed for future use case. Egress NAT44 is a common use case._
- Allow CNFs using peering protocols like Diameter to function by allowing connections to originate in either direction. _To support this capability will require session/protocol state tracking, that is beyond the scope of this GEP._
- Allow attaching policies (specifics to come later). _For example, attaching a TLS Policy to an Egress use case._ 

Commit: https://github.com/kubernetes-sigs/gateway-api/pull/1971/commits/4721decea6bd7e928e9428fe8ad2d86d3689aa48